File: 1-easy/problems/2614-prime-in-diagonal/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def diagonalPrime(self, nums):
        """
        :type nums: List[List[int]]
        :rtype: int
        """
        result = 0
        for i in range(len(nums)):
            if nums[i][i] in PRIMES_SET:
                result = max(result, nums[i][i])
            if nums[i][~i] in PRIMES_SET:
                result = max(result, nums[i][~i])
        return result

File: 1-easy/problems/2614-prime-in-diagonal/solution.ts
function diagonalPrime(nums: any): boolean | number | string | any {
  """
  :type nums: List[List[int]]
  :rtype: int
  """
  result = 0
  for (i in Array.from({length: (nums}, (_, i) => i).length)) {
      if (nums[i][i] in PRIMES_SET) {
          result = max(result, nums[i][i])
      if (nums[i][~i] in PRIMES_SET) {
          result = max(result, nums[i][~i])
  return result;
}

export { diagonalPrime };
